src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginCache.cs

```diff
@@ -46,6 +47,8 @@ public class TestPluginCache
         /// </summary>
         private AssemblyResolver assemblyResolver;
 
+        private IAssemblyResolver platformAssemblyResolver;
```

Comment:
Prefer `protected` since `internal` will enable other folks to directly use breaking the singleton usage.